### ISSUE TITLE
"Upload artifacts for environment" when using both Aurora and S3 in environment addons

### ISSUE THREAD

**Issue by richardbored** [aws/copilot-cli#4804]
I'm trying to use an Aurora instance and an S3 bucket at the environment level in my CoPilot app but I'm getting this error when I deploy the environment

`upload artifacts for environment production-env: parse environment addons: parameter logical ID "Env" defined in "db-env.yml" at Ln 6, Col 5 is different than in "s3-env.yml" at Ln 6, Col 5 `

Is this a bug in Copilot or have I done something wrong? If it is a bug, is there a workaround?

**Comment by Lou1415926** [comment]
Hey @richardbored ! Certainly an unusual error. Can you take a look at the `Parameters` section of `db-env.yml` and `s3-env.yml`, look for the parameter named `Env`? Does any of them has the `Default` field specified under? If you feel comfortable with sharing the snippet, it'd certainly help as well!

**Comment by richardbored** [comment]
I did find a workaround in the end. In `environments/addons/s3-env.yml`, I changed the Env Description from `The name of the environment being deployed.` to `The environment name your service, job, or workflow is being deployed to.` to match the description in `environments/addons/db-env.yml`.

Just to be clear, this is what Copilot produced for the s3-env.yml.
```
Parameters:
  App:
    Type: String
    Description: Your application's name.
  Env:
    Type: String
    Description: The name of the environment being deployed.
```

This is what I changed it to.

```
Parameters:
  App:
    Type: String
    Description: Your application's name.
  Env:
    Type: String
    Description: The environment name your service, job, or workflow is being deployed to.
```

I'm not entirely sure why that worked, but the env updated without any problems afterwards.

**Comment by Lou1415926** [comment]
Ahhh @richardbored great find!! 

> I'm not entirely sure why that worked, but the env updated without any problems afterwards.

When deploying addons, Copilot merges all the addon templates into one template and deploy it as a nested stack to your main service stack. 

Copilot merges the template section by section. So it requires that each template does not have conflicting information. For example you can't merge the following:
```yaml
# addons/a.yml
Parameters:
  Env:
     Type: String
      Default: test
# addons/b.yml
Parameters:
  Env:
     Type: String
      Default: prod
```

I will send out a fix to update the generated addons template so that each parameters' `Description` is the same. Thank you very much for pointing this out!

**Comment by KollaAdithya** [comment]
Fix is now released in v1.28 https://github.com/aws/copilot-cli/releases/tag/v1.28.0. Thanks @richardbored for reporting this!

**Comment by xxRockOnxx** [comment]
I would like to request to re-open this issue because it still exists.

The fix (making description the same) still works.
